### PR TITLE
feat(ui-shell): add platform side nav

### PR DESCRIPTION
**Pull request by joshblack**
Closes https://ibm.biz/BdYmbm

Add in experimental Platform Side Navigation.

#### Changelog

**New**

- `ui-shell`
  - Now includes Platform Side-nav components

**Changed**

- `_css--helpers.scss` now aliases `bx--assistive-text` to `visually-hidden`
- `ui-shell`
  - Updated hbs template and config to have variants for sidebar states

**Removed**

#### Testing / Reviewing

Can use the theme switcher to view the experimental work 👍 Goal is to have parity with the design files for this initial set of states. Will grow over time as we start getting into specific details.

### CODE REVIEW

File: src/globals/scss/_helper-classes.scss [hunk 1]
```diff
@@ -16,3 +16,16 @@
   white-space: nowrap;
   overflow: hidden;
 }
+
+.#{$prefix}--assistive-text {
+  position: absolute;
```
Comment (asudoh) [inline src/globals/scss/_helper-classes.scss]: Seems to have the same content as `.bx--visually-hidden`. Do you want to make one alias of another, or do you want to simply use `.bx--visually-hidden`?

Comment (joshblack) [reply src/globals/scss/_helper-classes.scss]: Oh duh, that's what it was. Couldn't remember the class name for the life of me haha. I'm down to do an alias here, I like the `assistive-text` convention after seeing it in lightning. Helps indicate what a node is potentially used for, at least in some of the cases here like the footer button text.

**Comment by joshblack** [comment]
cc @lovemecomputer

**Review by alisonjoseph — APPROVED** [review]
Overall looks good, question about the `get-fixed-size` should we move that in to a global file somewhere to use on all the experimental components we are bringing over?

Also, do we want to use get-fixed-size for spacing or the spacing variables? Should the spacing variables map to get-fixed-size? Thinking outloud.

**Review by aledavila — APPROVED** [review]
LGTM just a question do we want to keep the colors in theme for consistency?

**Comment by joshblack** [comment]
@alisonjoseph good question, I'm honestly not sure. I think we should be able to represent things using the spacing tokens but right now there doesn't seem to be a trend with it. We should bring this up to design, I think 👍 

@aledavila good point, definitely can refactor that way if that works 👍

**Review by asudoh — APPROVED** [review]
LGTM :+1: - Great to move the colors to theme as @aledavila pointed out. Thanks @joshblack!

**Pull request merged**